src/rust/ensogl/src/application/shortcut.rs

```diff
@@ -128,52 +154,93 @@ impl Handle {
         Self {rule}
     }
 
-    fn downgrade(&self) -> Instance {
+    fn downgrade(&self) -> WeakHandle {
         let rule = Rc::downgrade(&self.rule);
-        Instance {rule}
+        WeakHandle {rule}
     }
 }
 
+/// Weak version of the `Handle`.
 #[derive(Clone,CloneRef,Debug)]
-struct Instance {
+pub struct WeakHandle {
     rule : Weak<Rule>
 }
 
-impl Instance {
+impl WeakHandle {
     fn upgrade(&self) -> Option<Handle> {
         self.rule.upgrade().map(|rule| Handle {rule})
     }
 }
 
 
 
+// ==============
+// === Action ===
+// ==============
+
+/// A type of a keyboard action.
+#[derive(Clone,Copy,Debug,Eq,Hash,PartialEq)]
+#[allow(missing_docs)]
+pub enum ActionType {
+    Press, Release
+}
+
+/// Keyboard action defined as `ActionType` and `KeyMask`, like "release key 'n'".
+#[derive(Clone,Copy,Debug)]
+#[allow(missing_docs)]
+pub struct Action {
+    pub tp       : ActionType,
+    pub key_mask : KeyMask,
+}
```

Comment:
What is meaning of "releasing key_mask"? It is action of releasing all of keys in key_mask, or only a one of them? If I press Ctrl+Shift+S,and release, I can have three actions for releasing first S, then Shift, and Control?

Reply:
one of them. The key mask is the whole key binding, so releasing happens when it is no longer exactly in this state. Do you want better name or better docs?

Reply:
Better docs will be nice